tests/embeddings/test_elasticsearch.py

```diff
@@ -0,0 +1,39 @@
+import pytest
+from llama_index.embeddings.elasticsearch import ElasticsearchEmbeddings
+
+
+@pytest.fixture
+def model_id() -> str:
+    # Replace with your actual model_id
+    return "your_model_id"
+
+
+@pytest.fixture
+def es_url() -> str:
+    # Replace with your actual Elasticsearch URL
+    return "http://localhost:9200"
+
+
+@pytest.fixture
+def es_username() -> str:
+    # Replace with your actual Elasticsearch username
+    return "foo"
+
+
+@pytest.fixture
+def es_password() -> str:
+    # Replace with your actual Elasticsearch password
+    return "bar"
+
+
+def test_elasticsearch_embedding_constructor(
+    model_id: str, es_url: str, es_username: str, es_password: str
+) -> None:
+    """Test Elasticsearch embedding query."""
+
+    ElasticsearchEmbeddings.from_credentials(
```

Comment:
This test only runs if you have `pip install elasticsearch` taken place, so it's failing my testing locally.

Any chance @abstract829 you'd be willing to add a `pytest.mark.skipif` how `deeplake` does it: https://github.com/run-llama/llama_index/blob/v0.8.37/tests/indices/vector_store/test_deeplake.py#L108

Reply:
@jamesbraza how did it pass CI if it's failing?

Reply:
Yeah it's a good question.

`elasticsearch` is in `data_requirements.txt`, but not `requirements.txt`, so the fact it passes in CI is because CI also installs `data_requirements.txt`.  Is `data_requirements.txt` required?

I think `LlamaIndex` should be able to be tested from `setup.py`'s core `install_requires` and `pytest`, not require extras like `data_requirements.txt` to run.